core/src/main/java/org/apache/iceberg/hadoop/HadoopOutputFile.java

```diff
@@ -54,24 +58,53 @@ public static OutputFile fromPath(Path path, Configuration conf) {
     return fromPath(path, fs, conf);
   }
 
+  public static OutputFile fromPath(Path path, Configuration conf, Map<String, String> properties) {
+    short replicationFactor = -1;
+    if (properties != null) {
+      String replicationFactorAsString = properties.get(OutputFileFactory.FILE_REPLICATION_FACTOR);
+      if (replicationFactorAsString != null) {
+        replicationFactor = Short.parseShort(replicationFactorAsString);
+      }
+    }
+    return fromPath(path, conf, replicationFactor);
+  }
+
+  public static OutputFile fromPath(Path path, Configuration conf, short replication) {
+    FileSystem fs = Util.getFs(path, conf);
+    return new HadoopOutputFile(fs, path, conf, replication);
+  }
+
   public static OutputFile fromPath(Path path, FileSystem fs) {
     return fromPath(path, fs, fs.getConf());
   }
 
   public static OutputFile fromPath(Path path, FileSystem fs, Configuration conf) {
-    return new HadoopOutputFile(fs, path, conf);
+    return new HadoopOutputFile(fs, path, conf, (short) -1);
   }
 
-  private HadoopOutputFile(FileSystem fs, Path path, Configuration conf) {
+  private HadoopOutputFile(FileSystem fs, Path path, Configuration conf, short replication) {
     this.fs = fs;
     this.path = path;
     this.conf = conf;
+    this.replication = replication;
   }
 
   @Override
   public PositionOutputStream create() {
     try {
-      return HadoopStreams.wrap(fs.create(path, false /* createOrOverwrite */));
+      if (replication > 0) {
+        return HadoopStreams.wrap(
+            fs.create(
+                path,
+                false /* overwrite */,
+                conf.getInt(
+                    CommonConfigurationKeysPublic.IO_FILE_BUFFER_SIZE_KEY,
```

Comment:
why was this added?

Reply:
Added this change to wire-in the replication-factor as an argument to create file API. The additional params were the methods used by default implementation as part of  `FileSystem.create` 

```
create(f, overwrite, 
                  getConf().getInt("io.file.buffer.size", 4096),
                  getDefaultReplication(f),
                  getDefaultBlockSize(f)
```